doc: upmerge: Add a section for other zephyr updates (#2663)
nrfconnect/sdk-nrf: doc/nrf/releases/release-notes-latest.rst

  * Removed the ``net_idx`` parameter from the Health Client model APIs because it can be derived (by the stack) from the ``app_idx`` parameter.

> * Storage and filesystems:

Reviewer: ```suggestion
* Storage and file systems:
```

>   * A possible NULL pointer dereference when using any of the ``fs_`` functions has been fixed and will not return an error code instead.

Reviewer 2: Should this be "will now" instead of "will not"?

Author: Yes, thanks.

>   * A possible NULL pointer dereference when using any of the ``fs_`` functions has been fixed and will not return an error code instead.
>   * A garbage-collection issue has been fixed in the NVS subsystem.

Reviewer: ```suggestion
  * Fixed a possible NULL pointer dereference when using any of the ``fs_`` functions.
    The functions will not return an error code in this case anymore.
  * Fixed a garbage-collection issue in the NVS subsystem.
```

* Devicetree:

>   * All nRF-specific aliases with a dash (``-``) have been removed since they
>     are no longer needed (eg. ``i2c-0``, ``spi-0``, etc).

Reviewer: ```suggestion
  * Removed all nRF-specific aliases with a dash (``-``, for example, ``i2c-0`` or ``spi-0``), because they are no longer needed.
```

Reviewer 2: It's not the presence or absence of a dash which tells you if the alias was removed. Aliases that refer to a specific hardware peripheral instance were removed. The presence of a dash in the alias names is not directly relevant. This affects devicetree nodes for all peripheral instances, including SPI, I2C, UART, ADC, GPIO, GPIOTE, WDT, QDEC, RTC, TIMER, PWM, cryptocell, USB, FLASH, PDM, and EGU peripherals.

The aliases were defined because the legacy devicetree macros prevented users from referring to nodes by "node label", but allowed reference by alias.

The new devicetree.h API allows referring to nodes by node label, so the aliases were removed to avoid redundancy and confusion going forward.

For example, users should use `DT_NODELABEL(i2c0)` instead of `DT_ALIAS(i2c_0)`.

* Build system:

>   * The ``TEXT_SECTION_OFFSET`` symbol has been renamed to ``ROM_START_OFFSET``.
>   * A number of iterable section macros have been added to the set of linker macros, including ``Z_ITERABLE_SECTION_ROM`` and ``Z_ITERABLE_SECTION_RAM``.
>   * A new Zephyr Build Configuration package has been added, with support for specific build configuration for Zephyr derivatives (including forks). More info can be found `here <https://docs.zephyrproject.org/latest/guides/zephyr_cmake_package.html>`_.

Reviewer: ```suggestion
  * Renamed the ``TEXT_SECTION_OFFSET`` symbol to ``ROM_START_OFFSET``.
  * Added a number of iterable section macros to the set of linker macros, including ``Z_ITERABLE_SECTION_ROM`` and ``Z_ITERABLE_SECTION_RAM``.
  * Added a new Zephyr Build Configuration package with support for specific build configuration for Zephyr derivatives (including forks). 
    See :ref:`zephyr:cmake_pkg` for more information.
```

* Samples:

>   * The ``samples/boards/nrf/system_off`` sample has been updated to better support Nordic low-power states.
>   * The ``samples/subsys/usb/mass`` sample has been updated to perform all applicaiton-level configuration before the USB subsystem starts. The sample also supports FAT filesystems on external storage.

Reviewer: ```suggestion
  * Updated the :ref:`zephyr:nrf-system-off-sample` to better support low-power states of Nordic Semiconductor devices.
  * Updated the :ref:`zephyr:usb_mass` to perform all application-level configuration before the USB subsystem starts. 
    The sample now also supports FAT file systems on external storage.
```

Modules:

>   * A new ``depends`` keyword in a module's ``module.yml`` can now declare dependencies to other modules, in order to correctly establish the order of processing.

Reviewer: ```suggestion
  * Introduced a ``depends`` keyword that can be added to a module's :file:`module.yml` file to declare dependencies to other modules.
    This allows to correctly establish the order of processing.
```

Other:

>   * ``nanosleep`` has been implemented in the POSIX subsystem.
>   * The Zephyr-specific types have been deprecated in favor of the standard C99 int types.

Reviewer: ```suggestion
  * Implemented ``nanosleep`` in the POSIX subsystem.
  * Deprecated the Zephyr-specific types in favor of the standard C99 int types.
```

  * Removed the ``net_idx`` parameter from the Health Client model APIs because it can be derived (by the stack) from the ``app_idx`` parameter.

* Storage and file systems:

  * Fixed a possible NULL pointer dereference when using any of the ``fs_`` functions.
    The functions will now return an error code in this case.
  * Fixed a garbage-collection issue in the NVS subsystem.

* Devicetree:

>   * Removed all nRF-specific aliases to particular hardware peripherals, since they are no longer needed now that nodes are addressable by node label. As an example, you should now use ``DT_NODELABLE(i2c0)`` instead of ``DT_ALIAS(i2c_0)``.

Reviewer: ```suggestion
  * Removed all nRF-specific aliases to particular hardware peripherals, because they are no longer needed now that nodes can be addressed by node labels. 
    For example, you should now use ``DT_NODELABLE(i2c0)`` instead of ``DT_ALIAS(i2c_0)``.
```
And it's probably NODELABEL and not LABLE?
